### ISSUE TITLE
[client, protocol, node, card] Improve mnemonic security

### ISSUE THREAD

**Issue by ArjunBhuptani** [ConnextProject/indra#463]
There's two interconnected ways that we can improve the security of mnemonics in our system.

First, @bohendo's suggestion was to stop storing mnemonic in the store/db and have it be only used in-memory. This would involve a deep refactor to change how mnemonics are retrieved in the CF module in addition to a breaking change to how the CF module is instantiated (passing in mnemonic instead of store with mnemonic in it).

Second, we've discussed for some time that we need to add support for hosting the mnemonic OUTSIDE the Connext client and passing in a keygen fn which can spit out app-specific ephemeral keys as needed. This would involve a refactor of how keys are generated/retrieved in the CF module in addition to a breaking change to how the CF module is instantiated (to separate out passing the mnemonic exactly like above and to optionally support the keygen fn instead of mnemonic).

Note that the latter is needed for wallet integrations with Metamask, Mosendo, Torus, Portis, etc.  since none of them want to expose mnemonics to Connext code.

There's obviously a lot of overlap here so I think it makes sense to do this as one PR to the CF code altogether + a corresponding singular PR to ours. What do you guys think?

**Comment by rhlsthrm** [comment]
PR is blocked by #493

**Comment by rhlsthrm** [comment]
This is merged into master.
